src/sagas/identity.js

```diff
@@ -0,0 +1,136 @@
+import { all, takeLatest, call, put, select } from "redux-saga/effects";
+import {
+  // storeActiveIdentity,
+  getIdentities,
+  storeSeedIdentities,
+  storeSeedClaimCategories,
+  storeSeedClaims,
+  storeSeedAttestations,
+  getClaimCategories,
+  getClaims,
+  getAttestations,
+  updateAttestations,
+  updateIdentities,
+} from "../utils/asyncStore/identityStorage";
+import {
+  setActiveIdentity,
+  setIdentities,
+  setAttestations,
+  setClaims,
+  setClaimCategories,
+  storeIdentities,
+  addNewIdentity,
+  deselectActiveIdentity,
+  setNewActiveIdentity,
+} from "../actions/actionCreators";
+import { selectActiveIdentityId, selectAttestationsObject, selectIdentities, selectIdentityObj } from "../selectors/identity";
+import {
+  REQUEST_SEED_DATA,
+  SET_ACTIVE_IDENTITY,
+  STORE_IDENTITIES,
+  TOGGLE_ATTESTATION_PIN,
+  ADD_NEW_IDENTITY_NAME,
+  ADD_NEW_IDENTITY,
+  CHANGE_ACTIVE_IDENTITY,
+} from "../utils/constants/storeType";
+import {
+  normalizedCategories,
+  normalizedClaims,
+  normalizedAttestations,
+  normalizedIdentities,
+} from "../utils/identityTransform/identityNormalizers";
+import { denormalizedAttestations, denormalizedIdentities } from '../utils/identityTransform/identityDenormalizers';
+
+export default function* identitySaga() {
+  yield all([
+    takeLatest(REQUEST_SEED_DATA, handleSeedData),
+    takeLatest(STORE_IDENTITIES, handleStoreIdentities),
+    takeLatest(SET_ACTIVE_IDENTITY, handleSetActiveIdentity),
+    takeLatest(TOGGLE_ATTESTATION_PIN, handleToggleAttestation),
+    takeLatest(ADD_NEW_IDENTITY_NAME, handleAddNewIdentity),
+    takeLatest(ADD_NEW_IDENTITY, updateIdentityStorage),
+    takeLatest(CHANGE_ACTIVE_IDENTITY, handleChangeActiveIdentity),
+  ]);
+}
+
+function* handleSeedData() {
+  const savedIdentities = yield call(getIdentities);
+  if (!savedIdentities) {
+    yield call(storeSeedIdentities);
+  }
+  yield put(storeIdentities());
+}
+
+function* handleStoreIdentities() {
+  const storedIdentities = yield call(getIdentities);
+  const identities = yield call(normalizedIdentities, storedIdentities);
+  yield put(setIdentities(identities));
+  const selectedItentities = yield select(selectIdentities)
+  const activeIdentity = selectedItentities.find((identity) => identity.get('active') === true);
+  if (activeIdentity) {
+    yield put(setActiveIdentity(activeIdentity));
+  }
+}
+
+function* handleSetActiveIdentity() {
+  const selectedIdentityId = yield select(selectActiveIdentityId);
+    yield all([
+      call(storeSeedClaimCategories, selectedIdentityId),
```

Comment:
Seeding data needs to be independent of the logic you use to populate your store on a normal startup

Here:

start app first time
dispatch(appSetup())
takeLatest('APP_SETUP', handleAppSetup)
handleAppSetup = () => {
    const dataSeeded = yield call(fetchPropertyFromAsyncStoreThatIndicatesDataIsSeeded)
    if !dataSeeded {
       return put(seedData)
    }
    yield put(appSetupCompleted)
}

takeLatest('SEED_DATA', handleSeedData)

handleSeedData = () => {
    take all your consts and drop them into async storage
    this is async so wait for it to complete, then dispatch
    yield put(appSetupCompleted)
}

Now continue to login flow
In the login flow, you don't care about the seed data anymore. Now you just need to load the data from asyncStorage, and store it into your store.